lib/requestContextPlugin.js

```diff
@@ -9,12 +10,27 @@ const requestContext = {
 function plugin(fastify, opts, next) {
   fastify.decorate('requestContext', requestContext)
   fastify.decorateRequest('requestContext', requestContext)
+  const hook = opts.hook || 'onRequest'
 
-  fastify.addHook('onRequest', (req, res, done) => {
+  fastify.addHook(hook, (req, res, done) => {
     als.runWith(() => {
-      done()
+      const asyncResource = new AsyncResource('fastify-request-context')
+      req.asyncResource = asyncResource
```

Comment:
I would probably store this in a symbol instead

Reply:
Done